src/lib/Scenes/Home/Components/Sales/index.tsx

```diff
@@ -1,114 +1,91 @@
+  contentContainer: {
+    justifyContent: "space-between",
+    padding: 10,
+    display: "flex",
+  },
+})
 
-  renderList(itemData) {
-    const numColumns = Dimensions.get("window").width > 700 ? 4 : 2
-    return (
-      <FlatList
-        contentContainerStyle={{ justifyContent: "space-between", padding: 5, display: "flex" }}
-        data={itemData.data}
-        numColumns={numColumns}
-        keyExtractor={(item, index) => item.__id}
-        renderItem={d => {
-          return (
-            <TouchableWithoutFeedback onPress={this.handleTap.bind(this, d)}>
-              <View style={{ marginRight: 10, marginBottom: 10 }}>
-                <SaleItem key={d.index} sale={d.item} />
-              </View>
-            </TouchableWithoutFeedback>
-          )
-        }}
-      />
-    )
+class Sales extends React.Component<Props> {
+  get data() {
+    const { sales } = this.props
+    const liveAuctions = sales.filter(a => !!a.live_start_at)
+    const timedAuctions = sales.filter(a => !a.live_start_at)
+
+    return {
+      liveAuctions,
+      timedAuctions,
+    }
   }
 
   render() {
-    const sales = this.props.sales
-    const liveAuctions = sales.filter(a => !!a.live_start_at)
-    const timedAuctions = sales.filter(a => !a.live_start_at)
     const sections = [
       {
-        data: [
-          {
-            data: liveAuctions,
-          },
-        ],
+        data: [{ data: this.data.liveAuctions }],
         title: "Current Live Auctions",
+        renderItem: itemData => <SaleList {...itemData} />,
       },
       {
-        data: [
-          {
-            data: timedAuctions,
-          },
-        ],
+        data: [{ data: this.data.timedAuctions }],
         title: "Current Timed Auctions",
+        renderItem: itemData => <SaleList {...itemData} />,
+      },
+      {
+        data: [{ data: [] }],
+        title: "Lots by Artists You Follow",
+        renderItem: ({ title }) => <LotsByFollowedArtists title={title} />,
       },
     ]
 
     return (
       <SectionList
-        contentContainerStyle={{
-          justifyContent: "space-between",
-          padding: 15,
-          display: "flex",
-        }}
+        contentContainerStyle={SectionListStyles.contentContainer}
         stickySectionHeadersEnabled={false}
         sections={sections}
         keyExtractor={(item, index) => item.id}
-        renderItem={itemData => {
-          return this.renderList(itemData.item)
-        }}
-        renderSectionHeader={({ section }) =>
-          <SectionHeader>
-            <SectionTitle>
-              {section.title}
-            </SectionTitle>
-          </SectionHeader>}
       />
     )
   }
 }
 
+export function SalesRenderer({ render }) {
+  return (
+    <QueryRenderer
+      environment={createEnvironment()}
+      query={graphql`
+        query SalesRendererQuery {
+          sales: sales(live: true, is_auction: true) {
+            ...Sales_sales
+          }
+        }
```

Comment:
For the above reasons I would prefer to keep the query renderers next to each other for now, as it makes it easier to do the right thing (which is why they are all next to each other right now).

After we’ve shipped Eigen v4 and there’s better understanding of all the considerations that go into the current setup, we can revisit nicer layouts, but let’s not do so in these last moments towards the release 🙏 

FWIW The Relay Modern branch in Reaction already takes some steps towards making this more modular, whatever solution we settle on should be shared between the projects. https://github.com/artsy/reaction/commit/0878b658e3b4387b952cd4d47e428ee7028ca9c7, https://github.com/artsy/reaction/commit/7719f60075d615b4eddbbccfb26f862caf068a6c, https://github.com/artsy/reaction/commit/cd15a2f531a889c0679f7c055bae701d172184fe.

Reply:
I understand the concern of introducing new patterns before a huge launch but we've been colocating `QueryRenderers` closer to the components using them for several weeks now. For instance, the [`SalesRenderer`](https://github.com/artsy/emission/pull/796/files#diff-cf380933d958ee5caa27ff1dedb07da3) in the new Home view or  [`CategoriesRenderer`](https://github.com/artsy/emission/pull/824/files#diff-fd94edbe9875f44457f7c586634a723d) in the Saves & Follows view are in modules located in the same directory structure as the components using it.

Whether all `QueryRenderer` classes are in the same file or not, we'll still have to make sure we're not nesting them and thus avoid making several requests to render a single view. We'll also have to make sure they're using the same environment instance which can be easily done by exporting a `defaultEnvironment` from `lib/relay/createEnvironment`

Reply:
Ahh ok, that’s my bad then for not having fully internalised that earlier. Cool, yeah then never mind me on the aspect of colocating (which I absolutely definitely prefer in general!) 👍

Reply:
So to be clear, when one passes a createEnvironment instance from one QueryRender to the next it _does not_ optimize and merge requests? But sure thing, will move LotsByFollowedArtists to SaleRenderer.

Reply:
Correct. Each mounted query renderer results in a fetch https://facebook.github.io/relay/docs/query-renderer.html

The point of sharing an environment is to share cached data and rerender components that use data that was subsequently updated (e.g. through a mutation).